packages/block/src/header.ts

```diff
@@ -477,6 +526,103 @@ export class BlockHeader {
     return this.number.isZero()
   }
 
+  private _requireClique(name: string) {
+    if (this._common.consensusAlgorithm() !== 'clique') {
+      throw new Error(`BlockHeader.${name}() call only supported for clique PoA networks`)
+    }
+  }
+
+  /* Hash for PoA clique blocks is created without the seal */
+  private cliqueHash() {
+    const raw = this.raw()
+    raw[12] = this.extraData.slice(0, this.extraData.length - CLIQUE_EXTRA_SEAL)
+    return rlphash(raw)
+  }
+
+  /**
+   * Checks if the block header is an epoch transition
+   * header (only clique PoA, throws otherwise)
+   */
+  cliqueIsEpochTransition(): boolean {
+    this._requireClique('cliqueIsEpochTransition')
+    const epoch = new BN(this._common.consensusConfig().epoch)
+    // Epoch transition block if the block number has no
+    // remainder on the division by the epoch length
+    return this.number.mod(epoch).isZero()
+  }
+
+  /**
+   * Returns extra vanity data
+   * (only clique PoA, throws otherwise)
+   */
+  cliqueExtraVanity(): Buffer {
+    this._requireClique('cliqueExtraVanity')
+    return this.extraData.slice(0, CLIQUE_EXTRA_VANITY)
+  }
+
+  /**
+   * Returns extra seal data
+   * (only clique PoA, throws otherwise)
+   */
+  cliqueExtraSeal(): Buffer {
+    this._requireClique('cliqueExtraSeal')
```

Comment:
We can possible write these fancy `_requireClique` stuff as a [decorator](https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators). I have never used this before, but looks like these decorators are meant for exactly this!

Reply:
great idea! I will do that

Reply:
ok so I tried to implement this but unfortunately the decorator is executed when the class is defined, not instantiated, so it’s not possible to check against the set common for something like `@requireConsensus('clique')`